mteb/abstasks/TaskMetadata.py

```diff
@@ -386,5 +390,38 @@ def intext_citation(self, include_cite: bool = True) -> str:
             return f"\\cite{{{cite}}}"
         return cite
 
+    @property
+    def descriptive_stats(self) -> dict[str, DescriptiveStatistics] | None:
+        """Return the descriptive statistics for the dataset."""
+        if self.descriptive_stat_path.exists():
+            with self.descriptive_stat_path.open("r") as f:
+                return json.load(f)
+        return None
+
+    @property
+    def descriptive_stat_path(self) -> Path:
+        """Return the path to the descriptive statistics file."""
+        descriptive_stat_base_dir = Path(__file__).parent.parent / "descriptive_stats"
+        if not descriptive_stat_base_dir.exists():
+            descriptive_stat_base_dir.mkdir()
+        task_type_dir = descriptive_stat_base_dir / self.type
+        if not task_type_dir.exists():
+            task_type_dir.mkdir()
+        return task_type_dir / f"{self.name}.json"
+
+    @property
+    def n_samples(self) -> dict[str, int] | None:
+        """Returns the number of samples in the dataset"""
+        stats = self.descriptive_stats
+        if not stats:
+            return None
+
+        n_samples = {}
+        for subset, subset_value in stats.items():
+            if subset == "hf_subset_descriptive_stats":
+                continue
+            n_samples[subset] = subset_value["num_samples"]
+        return n_samples
```

Comment:
delete? Any reason to have it on its own?

Reply:
I've added this to simplify usage of `n_samples` in `docs/create_tasks_table.py`. Maybe in other parts of code this can be helpful

Reply:
let us keep it

Reply:
Should I change something?

Reply:
No, I think your argument is good - let us keep it as is.